tools/SendBlobs/FundsDistributor.cs

```diff
@@ -0,0 +1,192 @@
+using Org.BouncyCastle.Utilities.Encoders;
+
+namespace SendBlobs;
+internal class FundsDistributor
+{
+    /// <summary>
+    /// Distribute the available funds from an address to a number of newly generated private keys.
+    /// </summary>
+    /// <param name="nodeManager"></param>
+    /// <param name="chainId"></param>
+    /// <param name="distributeFrom"></param>
+    /// <param name="keysToMake"></param>
+    /// <param name="keyFilePath">Path to a file </param>
+    /// <returns><see cref="IEnumerable{string}"/> containing all the executed tx hashes.</returns>
+    /// <exception cref="Exception"></exception>
+    public async static Task<IEnumerable<string>> DitributeFunds(INodeManager nodeManager, ulong chainId, Signer distributeFrom, uint keysToMake, string keyFilePath, UInt256 maxFee, UInt256 maxPriorityFee)
+    {
+        string? balanceString = await nodeManager.Post("eth_getBalance", distributeFrom.Address, "latest");
+        if (balanceString is null)
+            throw new Exception($"Unable to get balance for {distributeFrom.Address}");
+        string? nonceString = await nodeManager.Post<string>("eth_getTransactionCount", distributeFrom.Address, "latest");
+        if (nonceString is null)
+            throw new Exception($"Unable to get nonce for {distributeFrom.Address}");
+
+        string? gasPriceRes = await nodeManager.Post<string>("eth_gasPrice") ?? "1";
+        UInt256 gasPrice = (UInt256)Convert.ToUInt64(gasPriceRes, gasPriceRes.StartsWith("0x") ? 16 : 10);
+
+        string? maxPriorityFeePerGasRes;
+        UInt256 maxPriorityFeePerGas = maxPriorityFee;
+        if (maxPriorityFee == 0)
+        {
+            maxPriorityFeePerGasRes = await nodeManager.Post<string>("eth_maxPriorityFeePerGas") ?? "1";
+            maxPriorityFeePerGas = (UInt256)Convert.ToUInt64(maxPriorityFeePerGasRes, maxPriorityFeePerGasRes.StartsWith("0x") ? 16 : 10);
+        }
+
+        UInt256 balance = new UInt256(Bytes.FromHexString(balanceString));
+
+        if (balance == 0)
+            throw new Exception($"Balance on provided signer {distributeFrom.Address} is 0.");
+
+        ulong nonce = Convert.ToUInt64(nonceString, nonceString.StartsWith("0x") ? 16 : 10);
+
+        UInt256 approxGasFee = (gasPrice + maxPriorityFeePerGas) * GasCostOf.Transaction;
+
+        //Leave 10% of the balance as buffer in case of gas spikes
+        UInt256 balanceMinusBuffer = (balance * 900) / 1000;
+        UInt256 totalFee = approxGasFee * keysToMake;
+
+        if (balanceMinusBuffer <= totalFee)
+            throw new Exception($"Not enough balance on {distributeFrom.Address} to distribute to {keysToMake} addresses");
+
+        UInt256 balanceToDistribute = balanceMinusBuffer - totalFee;
+
+        UInt256 perKeyToSend = balanceToDistribute / keysToMake;
+
+        using PrivateKeyGenerator generator = new();
+        IEnumerable<PrivateKey> privateKeys = Enumerable.Range(1, (int)keysToMake).Select(i => generator.Generate()).ToArray();
```

Comment:
Drop `ToArray` it is not needed

Reply:
After the refactor it will only be enumerated once, so now it is not needed :)